test/framework/src/main/java/org/elasticsearch/test/junit/listeners/ReproduceInfoPrinter.java

```diff
@@ -91,9 +91,20 @@ public void testFailure(Failure failure) throws Exception {
         GradleMessageBuilder gradleMessageBuilder = new GradleMessageBuilder(b);
         gradleMessageBuilder.appendAllOpts(failure.getDescription());
 
+        if (isRestApiCompatibilityTest(task)) {
+            b.append(System.lineSeparator());
+            b.append(
+                "This is a Rest Api Compatibility Test. "
+                    + "See the developers guide for details how to troubleshoot - REST_API_COMPATIBILITY.md"
+            );
+        }
         printToErr(b.toString());
     }
 
+    private boolean isRestApiCompatibilityTest(String task) {
+        return task.matches(".*yamlRestTestV\\dCompatTest");
```

Comment:
This is a bit brittle. Perhaps we can use an explicit system property here instead and then inject it in the rest compatibilty gradle plugin.

Reply:
do we do this for some other tasks?
You mean just setting with System.setProperty in YamlRestCompatTestPlugin ? 
we have to remember to remove it when the tasks finishes. Does this not prevent concurrent task runs?

Reply:
Yes, we use the same pattern for [identifying if this is a BWC test](https://github.com/elastic/elasticsearch/blob/0832232b997fa0e50f8dcf1911307a2e8771a77f/test/framework/src/main/java/org/elasticsearch/test/junit/listeners/ReproduceInfoPrinter.java#L67). We can just add something like a `tests.restCompat` system property when we setup the test task [here](https://github.com/elastic/elasticsearch/blob/0ec229050e5bebdb82a0a273076aae52e56ba520/build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/rest/compat/YamlRestCompatTestPlugin.java#L198-L216).